agent/check_handler.go

```diff
@@ -223,3 +224,44 @@ func extractMetrics(event *types.Event) []*types.MetricPoint {
 
 	return transformer.Transform()
 }
+
+// mergeEnvironments merges env2 into env1, overwriting any existing variable
+// in env1, except for the "special" variables PATH, CPATH and LD_LIBRARY_PATH.
+// These variables are merged by prepending the value from env2 to the value in
+// env1, effectively giving priority to the value from env2.
+func mergeEnvironments(env1, env2 []string) []string {
+	toMap := func(s []string) map[string]string {
+		m := map[string]string{}
+
+		for _, v := range s {
+			split := strings.SplitN(v, "=", 2)
+			m[split[0]] = split[1]
+		}
+
+		return m
+	}
+
+	fromMap := func(m map[string]string) []string {
```

Comment:
Similar to the above closure, this function doesn't really use anything it closes over, so why not make it a stand-alone?